virtcontainers/qemu.go

```diff
@@ -610,7 +636,8 @@ func (q *qemu) waitSandbox(timeout int) error {
 		return err
 	}
 
-	return nil
+	timeoutDuration := time.Duration(timeout)*time.Second - time.Now().Sub(timeStart)
+	return q.waitAgent(eventCh, timeoutDuration)
```

Comment:
Hi @ningbo9, I'm not sure about the issue you're trying to solve here. I have checked the description of `boot_delay` which says `Milliseconds to delay each printk during boot.`. And because every printk gets delayed, adding a simple `boot_delay=1` bumps by 3s the container boot time. If I do `boot_delay=2`, I get a bump around 6s. So clearly putting `boot_delay=100` (~ 5min) will exceed/reach any timeout we might have inside Kata Containers right now.

I know you took `boot_delay` as an example to reproduce the behavior you were seeing when moving to a slower hypervisor, but I'm wondering if we even want to support such cases where it takes ages to the VM to be created, hence the serial port to also be created?